### PR TITLE
TRUNK-5127: Deprecate unused method and field

### PR DESCRIPTION
**Pull request by bmiddle**
## Description of what I changed

After a few revisions, I have added on a javadoc comment explaining the version of which the method and field were deprecated. There does not seem to be a replacement available, so that is also mentioned in the javadoc.
I added the `@Deprecated` decorator to the variable `WebUtil.defaultDateCache` and to the function `WebUtil.setDefaultDateCache(String)`. It seems that these two are not used within the OpenMRS system, but since there is the possibility that other modules may use these I have deprecated these two.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5127

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: web/src/main/java/org/openmrs/web/WebUtil.java [hunk 1]
```diff
@@ -30,7 +30,8 @@
 public class WebUtil implements GlobalPropertyListener {
 	
 	private static final Logger log = LoggerFactory.getLogger(WebUtil.class);
-	
+
+	@Deprecated
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/WebUtil.java]: Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation

File: web/src/main/java/org/openmrs/web/WebUtil.java [hunk 2]
```diff
@@ -419,7 +420,8 @@ public static String formatDate(Date date, Locale locale, FORMAT_TYPE type) {
 	public boolean supportsPropertyName(String propertyName) {
 		return OpenmrsConstants.GP_SEARCH_DATE_DISPLAY_FORMAT.equals(propertyName);
 	}
-	
+
+	@Deprecated
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/WebUtil.java]: Same question as above

**Comment by bmiddle** [comment]
@dkayiwa, I amended the commit and it should reflect the convention well. Thank you for pointing me in the right direction

File: web/src/main/java/org/openmrs/web/WebUtil.java [hunk 3]
```diff
@@ -30,7 +30,11 @@
 public class WebUtil implements GlobalPropertyListener {
 	
 	private static final Logger log = LoggerFactory.getLogger(WebUtil.class);
-	
+
+	/**
+	 * @deprecated as of 2.2.0-SNAPSHOT, there is no replacement field available.
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/WebUtil.java]: That should be the version to be released. And hence it is 2.2.0

Comment (bmiddle) [reply web/src/main/java/org/openmrs/web/WebUtil.java]: I will get this fixed

Comment (bmiddle) [reply web/src/main/java/org/openmrs/web/WebUtil.java]: Should be fixed now

**Comment by bmiddle** [comment]
@dkayiwa, will this work?

File: web/src/main/java/org/openmrs/web/WebUtil.java [hunk 4]
```diff
@@ -30,7 +30,11 @@
 public class WebUtil implements GlobalPropertyListener {
 	
 	private static final Logger log = LoggerFactory.getLogger(WebUtil.class);
-	
+
+	/**
+	 * @deprecated as of 2.2.0, there is no replacement field available.
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/WebUtil.java]: Do we need this for the private field declaration?

**Comment by dkayiwa** [comment]
@bmiddle are you still working on this?

**Comment by bmiddle** [comment]
@dkayiwa we are no longer working on this ticket so it should be okay to close the PR and unassign the ticket

**Pull request closed without merging**